### PR TITLE
Feature/test newsletters' html generation

### PR DESCRIPTION
**Pull request by javierriveros**

### CODE REVIEW

File: test_project/templates/newsletters/base_hubble.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    {% include "newsletters/includes/preferences_top.html" %}
+</table>
+<!-- // End Module: Standard Preheader \ -->
+{% endblock %}
+
+
+{% block header %}
+    {% with width="700" height="300" color="#000000" logo_width="140" logo_url=base_url|add:MEDIA_URL|add:"esa_nl_logo.png" %}
+    {% block header_content %}
+        {% with header_title="ESA/Hubble" %}
```
Comment (EdisonArango) [inline test_project/templates/newsletters/base_hubble.html]: Please replace all `ESA/Hubble` references

File: djangoplicity/newsletters/templates/newsletters/newsletter_detail.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{% extends 'base.html' %}
+{% extends 'newsletters/base.html' %}
```
Comment (EdisonArango) [inline djangoplicity/newsletters/templates/newsletters/newsletter_detail.html:1]: Why was this necessary?.

Comment (javierriveros) [reply djangoplicity/newsletters/templates/newsletters/newsletter_detail.html:1]: Sorry, this was causing me an error with the previous test data that I was using, I'll remove it right away

Comment (javierriveros) [reply djangoplicity/newsletters/templates/newsletters/newsletter_detail.html:1]: Well, I just run the tests after removing it and it is causing me an error
image

Comment (EdisonArango) [reply djangoplicity/newsletters/templates/newsletters/newsletter_detail.html:1]: Ok, then I believe it's actually a bug. 👍

File: test_project/templates/newsletters/news-text.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{% load djangoplicity_text_utils release_links djangoplicity_datetime %}
+{% with announcements=data.announcements potws=data.potws newon=data.newonhubble org="Djangoplicity Newsletters" embargo="" reason="you subscribed to Djangoplicity Newsletters News" email="hubble@eso.org" %}{% with signature_date=data.release.release_date signature1="Djangoplicity Newsletters Information Centre" %}{% include "newsletters/includes/text/press_release.txt" %}{% endwith %}
```
Comment (EdisonArango) [inline test_project/templates/newsletters/news-text.html]:
```suggestion
{% with announcements=data.announcements potws=data.potws newon=data.newonhubble org="Djangoplicity Newsletters" embargo="" reason="you subscribed to Djangoplicity Newsletters News" email="test@example.com" %}{% with signature_date=data.release.release_date signature1="Djangoplicity Newsletters Information Centre" %}{% include "newsletters/includes/text/press_release.txt" %}{% endwith %}
```

File: test_project/templates/newsletters/science-text.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+{{ announcement.release_date|date }}: {{ announcement.description|striptags|truncatewords:50|unescape|safe }} [read more on {{base_url}}{{ announcement.get_absolute_url }}]
+
+
+
+{% comment %} How to advertise your Hubble results: {% endcomment %}
+{% comment %} https://www.spacetelescope.org/forscientists/announcements/sci16008/ {% endcomment %}
+
+{% endfor %}{% endif %}
+{% with email="djangoplicity-newsletters@eso.org" %}{% include "newsletters/includes/text/footer.txt" %}{% endwith %}
```
Comment (EdisonArango) [inline test_project/templates/newsletters/science-text.html]:
```suggestion
{% with email="djangoplicity-newsletters@example.com" %}{% include "newsletters/includes/text/footer.txt" %}{% endwith %}
```

File: test_project/templates/newsletters/science.html [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    <br /><hr style="{{hr_style}}"><br />
+    {% with objects=data.scienceannouncements org="" title="Science Announcements" urlpattern="scienceannouncements_defaultquery" truncatelength=60 %}
+        {% include "newsletters/includes/archiveobjects.html" with width_right="100%" %}
+    {% endwith %}
+    {% endif %}
+
+{% endblock %}
+
+{% block footer %}
+{% with org="Djangoplicity Newsletters" potw_feed_url="http://feeds.feedburner.com/hubble_potw/" reason="you subscribed to the Djangoplicity Newsletters Science Newsletter" footer_potws=data.footer_potws footer_note='' unsubscribe_link='' preferences_link='' browser_link=base_url|add:"/newsletters/science/html/"|add:newsletter_id|add:"/" address="ESA/Hubble c/o ESO, Karl-Schwarzschild-Str 2, D-85748 Garching bei M&uuml;nchen, Germany" %}
```
Comment (EdisonArango) [inline test_project/templates/newsletters/science.html]:
```suggestion
{% with org="Djangoplicity Newsletters" potw_feed_url="http://feeds.feedburner.com/hubble_potw/" reason="you subscribed to the Djangoplicity Newsletters Science Newsletter" footer_potws=data.footer_potws footer_note='' unsubscribe_link='' preferences_link='' browser_link=base_url|add:"/newsletters/science/html/"|add:newsletter_id|add:"/" address="3055 Street Name, Moon, MW 78111" %}
```

File: tests/test_email_generation.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, self.newsletter.editorial_text)
+
+    def test_newsletter_detail_html(self):
+        """Test newsletter html is properly rendered"""
+        url = '/newsletters/%s/html/%s/' % (self.newsletter_type.slug, self.newsletter.pk)
+        response = self.client.get(url, follow=True)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, """<iframe src="/newsletters/{0}/htmlembed/{1}/" width="100%" scrolling="no" frameBorder="0" onload='javascript:resizeIframe(this);'></iframe>""".format(self.newsletter_type.slug, self.newsletter.pk))
```
Comment (EdisonArango) [inline tests/test_email_generation.py:50]: Cool 👍

File: test_project/templates/newsletters/science.html [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    <br /><hr style="{{hr_style}}"><br />
+    {% with objects=data.scienceannouncements org="" title="Science Announcements" urlpattern="scienceannouncements_defaultquery" truncatelength=60 %}
+        {% include "newsletters/includes/archiveobjects.html" with width_right="100%" %}
+    {% endwith %}
+    {% endif %}
+
+{% endblock %}
+
+{% block footer %}
+{% with org="Djangoplicity Newsletters" potw_feed_url="http://feeds.feedburner.com/hubble_potw/" reason="you subscribed to the Djangoplicity Newsletters Science Newsletter" footer_potws=data.footer_potws footer_note='' unsubscribe_link='' preferences_link='' browser_link=base_url|add:"/newsletters/science/html/"|add:newsletter_id|add:"/" address="Djangoplicity Newsletters c/o ESO, Karl-Schwarzschild-Str 2, D-85748 Garching bei M&uuml;nchen, Germany" %}
```
Comment (EdisonArango) [inline test_project/templates/newsletters/science.html]: Please change the address to a Fake one

File: tests/test_email_generation.py (hunk 6 above)
Comment (javierriveros) [reply tests/test_email_generation.py:50]: I know it's not that much code, but it took me a lot longer than I expected 😅

**Comment by javierriveros** [comment]
I think I've solved all the suggestions, so let me know if you have any other suggestion

File: djangoplicity/newsletters/templates/newsletters/newsletter_detail.html (hunk 2 above)
Comment (javierriveros) [reply djangoplicity/newsletters/templates/newsletters/newsletter_detail.html:1]: Yeah, I'll leave this here for reference: https://github.com/djangoplicity/djangoplicity-newsletters/runs/1190341999

Comment (EdisonArango) [reply djangoplicity/newsletters/templates/newsletters/newsletter_detail.html:1]: This change caused an bad display of the newsletters for NOIRLab, I'm going to revert it.
